src/parquet/arrow/writer.h

```diff
@@ -50,8 +50,7 @@ class PARQUET_EXPORT FileWriter {
   FileWriter(::arrow::MemoryPool* pool, std::unique_ptr<ParquetFileWriter> writer);
 
   ::arrow::Status NewRowGroup(int64_t chunk_size);
-  ::arrow::Status WriteColumnChunk(
-      const ::arrow::Array* data, int64_t offset = 0, int64_t length = -1);
+  ::arrow::Status WriteColumnChunk(const ::arrow::Array* data);
```

Comment:
It's not in scope for this patch, but we should consider it before 1.0: this argument should really be `const Array&` instead of `const Array*` -- there's other places where `const T*` should be `const T&`. see https://google.github.io/styleguide/cppguide.html#Reference_Arguments

```
However, there are some instances where using const T* is preferable to const T& for input parameters. For example:

* You want to pass in a null pointer.
* The function saves a pointer or reference to the input.
```